### PR TITLE
Fix vimc-6544 - quote csv fields that need it

### PR DESCRIPTION
**Pull request by weshinsley**

### CODE REVIEW

**Comment by weshinsley** [comment]
https://mrc-ide.myjetbrains.com/youtrack/issue/VIMC-6544

File: R/utils.R [hunk 1]
```diff
@@ -1,5 +1,6 @@
-write_csv <- function(...) {
-  utils::write.csv(row.names = FALSE, quote = FALSE, ...)
+write_csv <- function(x, file) {
+  commas <- which(sapply(x, function(y) any(grepl(",",y))))
```
Comment (r-ash) [inline R/utils.R]: Could do this in type safe way
```
which(vapply(x, function(y) any(grepl(",", y), logical(1)))
```

**Review by r-ash — APPROVED** [review]
Looks good, one small comment

File: R/utils.R (hunk 1 above)
Comment (weshinsley) [reply R/utils.R]: Done - thanks!

**Pull request merged**